archunit/src/test/java/com/tngtech/archunit/lang/ArchRuleTest.java

```diff
@@ -159,6 +161,20 @@ public void finish(ConditionEvents events) {
         assertThat(condition.eventsFromFinish.getViolating()).hasSize(1);
     }
 
+    @Test
+    public void empty_input_throws_an_assertion_exception_by_default() {
+        thrown.expect(AssertionError.class);
```

Comment:
I always have a little bad feeling about testing `AssertionErrors` like that. Because that is also the error used by JUnit itself to mark tests as failed. What about using `try .. catch` here and also asserting the error message? Because at the moment this test would be green for any test failure whatsoever, no?

Reply:
Yes, good point. I extended the test, so that also the error message is asserted.